commands/keys/generate.go

```diff
@@ -0,0 +1,225 @@
+	privHeaderPath := keysKeychainDir.Join(privHeader + ".h")
+	if privHeaderPath.Exist() {
+		return errors.New(tr("File already exists: %s", privHeaderPath))
+	}
+	privHeaderFile, err := privHeaderPath.Create()
+	if err != nil {
+		return err
+	}
+	defer privHeaderFile.Close()
+
+	// create the public header files
+	pubHeader := strings.TrimSuffix(pubKeyPath.Base(), pubKeyPath.Ext())
+	pubHeaderPath := keysKeychainDir.Join(pubHeader + ".h")
+	if pubHeaderPath.Exist() {
+		return errors.New(tr("File already exists: %s", pubHeaderPath))
+	}
+	pubHeaderFile, err := pubHeaderPath.Create()
+	if err != nil {
+		return err
+	}
+	defer pubHeaderFile.Close()
+
+	privBlock := pem.Block{
+		Type:  privPemType,
+		Bytes: priv509,
+	}
+	err = pem.Encode(privKeyFile, &privBlock)
+	if err != nil {
+		return err
+	}
+	err = genCFile(privHeaderFile, priv509, "priv")
+	if err != nil {
+		return err
+	}
+
+	pubBlock := pem.Block{
+		Type:  pubPemType,
+		Bytes: pubAsn1,
+	}
+	err = pem.Encode(pubKeyFile, &pubBlock)
+	if err != nil {
+		return err
+	}
+	err = genCFile(pubHeaderFile, pubAsn1, "pub")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// genEcdsaP256 will generate private and public ecdsap256 keypair and return them
+// it will encode the private PKCS #8, ASN.1 DER form, and for the public will use the PKIX, ASN.1 DER form
+// adapted from https://git.furworks.de/Zephyr/mcuboot/src/commit/3869e760901a27adff47ccaea803a42f1b0169c0/imgtool/imgtool.go#L165
+func genEcdsaP256() ([]byte, []byte, error) {
+	priv, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	keyPriv, err := x509.MarshalPKCS8PrivateKey(priv)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pub := priv.Public()
+	keyPub, err := x509.MarshalPKIXPublicKey(pub)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return keyPriv, keyPub, nil
+}
+
+// genCFile will take a file as input, the byte slice that represents the key and the type of the key.
+// If will then write the file and report an error if one is found
+func genCFile(file *os.File, bytes []byte, t string) (err error) {
+	fileContent := fmt.Sprintf(`/* Autogenerated, do not edit */
+const unsigned char rsa_%s_key[] = {
+	%s
+};
+const unsigned int ec_%s_key_len = %d;
+`, t, formatCData(bytes), t, len(bytes))
+	_, err = file.WriteString(fileContent)
+	return err
+}
+
+// formatCData will take the byte slice representing a key and format correctly as "C" data. It will return it as a string
+// taken and adapted from https://git.furworks.de/Zephyr/mcuboot/src/commit/3869e760901a27adff47ccaea803a42f1b0169c0/imgtool/imgtool.go#L313
+func formatCData(data []byte) string {
+	buf := new(bytes.Buffer)
+	indText := strings.Repeat("\t", 1)
+	for i, b := range data {
+		if i%8 == 0 {
+			if i > 0 {
+				fmt.Fprintf(buf, "\n%s", indText)
+			}
+		} else {
+			fmt.Fprintf(buf, " ")
+		}
+		fmt.Fprintf(buf, "0x%02x,", b)
```

Comment:
```suggestion
  // clang-format off
%s
  // clang-format on
};
const unsigned int ec_%s_key_len = %d;
`, t, formatCData(bytes), t, len(bytes))
	_, err = file.WriteString(fileContent)
	return err
}

// formatCData will take the byte slice representing a key and format correctly as "C" data. It will return it as a string
// taken and adapted from https://git.furworks.de/Zephyr/mcuboot/src/commit/3869e760901a27adff47ccaea803a42f1b0169c0/imgtool/imgtool.go#L313
func formatCData(data []byte) string {
	buf := new(bytes.Buffer)
	indText := strings.Repeat(" ", 2)
	for i, b := range data {
		if i%8 == 0 {
			fmt.Fprintf(buf, indText)
		} else {
			fmt.Fprintf(buf, " ")
		}
		fmt.Fprintf(buf, "0x%02x,", b)
		if i%8 == 7 {
			fmt.Fprintf(buf, "\n")
		}
```

- Add [control comments](https://releases.llvm.org/11.0.1/tools/clang/docs/ClangFormatStyleOptions.html#disabling-formatting-on-a-piece-of-code) to prevent ClangFormat from breaking the initialization list to one element per line.
  - (I didn't figure out which of the configuration settings causes this behavior)
- Use official Arduino indentation style (two spaces)
  - https://github.com/arduino/arduino-language-server/blob/0.6.0/ls/ls_formatter.go#L98
  - https://github.com/arduino/arduino-language-server/blob/0.6.0/ls/ls_formatter.go#L158
- Make formatter code work for first line.